src/configure/templateInputHelper/InputControlProvider.ts

```diff
@@ -1,26 +1,36 @@
 import { GenericResource } from "azure-arm-resource/lib/resource/models";
+import { ControlProvider } from "../helper/controlProvider";
 import { MustacheHelper } from "../helper/mustacheHelper";
 import { ExtendedInputDescriptor, ExtendedPipelineTemplate, InputDataType, InputMode } from "../model/Contracts";
-import { AzureSession, ControlType, IPredicate, StringMap } from '../model/models';
+import { AzureSession, ControlType, IPredicate, RepositoryAnalysisApplicationSettings, StringMap } from '../model/models';
 import { InputControl } from "./InputControl";
 import { DataSourceExpression } from "./utilities/DataSourceExpression";
 import { InputControlUtility } from "./utilities/InputControlUtility";
 import { VisibilityHelper } from "./utilities/VisibilityHelper";
 
 export class InputControlProvider {
+    private readonly repoAnalysisSettingKey: string = "repoAnalysisSettingKey";
     private _pipelineTemplate: ExtendedPipelineTemplate;
     private _inputControlsMap: Map<string, InputControl>;
+    private _repoAnalysisSettings: RepositoryAnalysisApplicationSettings[];
+    private _repoAnalysisSettingInUse: number;
 
-    constructor(pipelineTemplate: ExtendedPipelineTemplate, context: { [key: string]: any }) {
+    constructor(pipelineTemplate: ExtendedPipelineTemplate, repoAnalysisSettings: RepositoryAnalysisApplicationSettings[], context: { [key: string]: any }) {
         this._pipelineTemplate = pipelineTemplate;
         this._inputControlsMap = new Map<string, InputControl>();
+        this._repoAnalysisSettings = repoAnalysisSettings || [];
```

Comment:
Can all repo analysis related fields and methods be moved to a separate class? this will clean IPC class cleaner.

Reply:
done